src/infrasys/time_series_models.py

```diff
@@ -102,7 +102,7 @@ def check_data(
         return data
 
     @classmethod
-    def aggregate(cls, ts_data: list[Self]) -> Self:
+    def aggregate(cls, ts_data: list[Self], agg_type: Literal["sum", "avg"] = "sum") -> Self:
```

Comment:
I think we should explore to have this method as a standalone function. It feels unnatural to me that we define a method that can take multiple instances of itself. I am open for discussion.